bin/check-netty-zookeeper-cluster.rb

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env ruby
+#  encoding: UTF-8
+
+#
+#  netty-check-zookeeper-cluster
+#
+# DESCRIPTION:
+#   Check if a Zookeeper 3.5 cluster is OK.
+#
+#   This check will each verify each node for errors,
+#   check averag latency for a threshold value,
+#   and if the cluster has a leader, make sure it has the right number of followers.
+#
+# PLATFORMS:
+#   All
+#
+# DEPENDENCIES:
+#   gem: sensu-plugin
+#
+# USAGE:
+#  Check if a node has Zookeeper running and responds with imok.
+#
+#  Cluster should have 3 (n+1) nodes and average latency threshold should be 10
+#  ./netty-check-zookeeeper-cluster.rb -c 2 -l 10
+#
+# LICENCE:
+#   Duncan Schulze <duschulze@gmail.com>
+#   Released under the same terms as Sensu (the MIT license); see LICENSE
+#   for details.
+#
+
+require 'sensu-plugin/check/cli'
+require 'rest-client'
+require 'json'
+
+class CheckZookeeperCluster < Sensu::Plugin::Check::CLI
+  option :netty_port,
+         description: 'Zookeeper nodes\' listen port',
+         short: '-p port',
+         long: '--port port',
+         default: 8080,
+         proc: proc(&:to_i)
+
+  option :followers,
+         description: 'Zookeeper cluster follower count',
+         short: '-c count',
+         long: '--count count',
+         default: 2,
+         proc: proc(&:to_i)
+
+  option :latency,
+         description: 'Critical threshold for Zookeeper average latency',
+         short: '-l TICKS',
+         long: '--latency TICKS',
+         proc: proc(&:to_i),
+         default: 10
+
+  def get_monitoring_output(port)
+    r = RestClient::Resource.new("http://localhost:#{port}/commands/monitor",
```

Comment:
I don't see the `rest-client` gem in the `.gemspec` we would need to add it there to ensure that the dependencies will be met at runtime. Anyways I'd prefer to use `Net::HTTP` as `rest-client` would add the dependency of a `c` compiler.